projects/packages/action-bar/src/class-action-bar.php

```diff
@@ -46,18 +47,111 @@ public function enqueue_scripts() {
 	 * Render app container html.
 	 */
 	public function print_html() {
-		if ( is_admin() || ! is_single() ) {
+		if ( ! $this->is_active() ) {
+			return;
+		}
+
+		$post_id = get_the_ID();
+
+		if ( ! is_numeric( $post_id ) ) {
 			return;
 		}
 
-		echo '<div id="jetpack-action-bar" class="jetpack-action-bar"></div>';
+		$protocol = is_ssl() ? 'https' : 'http';
+
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			$blog_id = get_current_blog_id();
+			// Use the mapped domain, if there is one.
+			$domain = get_primary_redirect( $blog_id );
+			$url    = "https://$domain";
+		} else {
+			$blog_id   = \Jetpack_Options::get_option( 'id' );
+			$url       = home_url();
+			$url_parts = wp_parse_url( $url );
+			$domain    = $url_parts['host'];
+		}
+
+		$blog_name  = get_bloginfo( 'name' );
+		$post_url   = get_permalink( $post_id );
+		$reader_url = $this->get_reader_url( $blog_id );
+		$widget_src = sprintf( 'https://widgets.wp.com/action-bar/#blog_id=%2$d&amp;post_id=%3$d&amp;origin=%1$s://%4$s', $protocol, $blog_id, $post_id, $domain );
```

Comment:
The domain should be urlencoded, just in case there's something odd going on.

As for the ampersands, it's probably better to let `esc_url()` do it later than to do them here.
```suggestion
		$widget_src = sprintf( 'https://widgets.wp.com/action-bar/#blog_id=%2$d&post_id=%3$d&origin=%1$s://%4$s', $protocol, $blog_id, $post_id, rawurlencode( $domain ) );
```